dashboard/app/models/levels/flappy.rb

```diff
@@ -0,0 +1,103 @@
+# == Schema Information
+#
+# Table name: levels
+#
+#  id                    :integer          not null, primary key
+#  game_id               :integer
+#  name                  :string(255)      not null
+#  created_at            :datetime
+#  updated_at            :datetime
+#  level_num             :string(255)
+#  ideal_level_source_id :integer
+#  user_id               :integer
+#  properties            :text(65535)
+#  type                  :string(255)
+#  md5                   :string(255)
+#  published             :boolean          default(FALSE), not null
+#  notes                 :text(65535)
+#  audit_log             :text(65535)
+#
+# Indexes
+#
+#  index_levels_on_game_id  (game_id)
+#  index_levels_on_name     (name)
+#
+
+class Flappy < Blockly
+  serialized_attrs %w(
+    success_condition
+    failure_condition
+    default_flap
+    free_play
+    ground
+    obstacles
+    score
+    shareable
```

Comment:
It's unfortunately pretty inconsistent; all level types consider the free_play flag when deciding whether or not to share, but they all have their own extra bits of logic, too. For flappy, sharing is determined by `level.freePlay && level.shareable`.